### PR TITLE
Security controls > Tokenization docs

### PR DESCRIPTION
**Pull request by iamnotacake**
* Describe what tokenization is
* Add Rust example

WIP:
- [X] Golang example of gRPC API usage
- [X] HTTP API usage example

Questions:
- What should we put in `acra/security-controls/_index.md`? The main reason this page exists is to give us access to sub-pages, but it cannot be empty or just contain "TODO".
  > Brief list of subpages.
- Tokeni**s**ation or tokeni**z**ation? I used the second one, but if the first one is better and used everywhere else I'll replace.
  > No problems with tokeni**z**ation.

### CODE REVIEW

**Comment by vixentael** [comment]
"tokenization" is fine

File: acra/security-controls/tokenisation/_index.md [hunk 1]
```diff
@@ -2,3 +2,170 @@
...
+Better see than read:
+
+* `789` -> `156749362`
+* `testing testing` -> `lwcIvM3zaJN9cbN`
+* `kate@example.com` -> `sVAOfOwF@mSuGPog.et`
+
+The tokenization process has few interesting properties:
+
+* Tokenized value is simply random, it is neither encrypted nor hashed version of the input
+* Tokenization of the same value more than once will return same result
```
Comment (Lagovas) [inline acra/security-controls/tokenisation/_index.md:22]: may return the same result (in consistent mode) or random (not consistent) even for the same value according to configuration

File: acra/security-controls/tokenisation/_index.md [hunk 2]
```diff
@@ -2,3 +2,170 @@
...
+* Tokenization can be reversed, but only for valid (previously returned) values
+
+## API
+
+Tokenization is done by Acra Translator.
```
Comment (Lagovas) [inline acra/security-controls/tokenisation/_index.md:27]: AcraServer tokenize values too according to `encryptor_config_file`

File: acra/security-controls/tokenisation/_index.md (hunk 1 above)
Comment (vixentael) [inline acra/security-controls/tokenisation/_index.md:22]:
```suggestion
* Tokenization supports two modes: tokenization of the same value will return the same result every time (called `"consistent tokenization"`). Tokenization of the same value will return different results every time (called `"inconsistent tokenization"`.
```

File: acra/security-controls/tokenisation/_index.md [hunk 3]
```diff
@@ -2,3 +2,170 @@
...
+## Description
+
+Tokenization is a way to transform some sensitive data into pseudo-random representation of same data type.
+A number will remain a number, string will be string, email will be email.
+But the tokenized version will look like a random value.
+Better see than read:
+
+* `789` -> `156749362`
+* `testing testing` -> `lwcIvM3zaJN9cbN`
+* `kate@example.com` -> `sVAOfOwF@mSuGPog.et`
```
Comment (storojs72) [inline acra/security-controls/tokenisation/_index.md:17]: It seems that email is not data type. I guess that tokenization targets not only data type but also format (email will be email). Should we elaborate on this in phrase above?

File: acra/security-controls/tokenisation/_index.md [hunk 4]
```diff
@@ -2,3 +2,170 @@
...
 ---
+
+# Tokenization
+
+## Description
+
+Tokenization is a way to transform some sensitive data into pseudo-random representation of same data type.
+A number will remain a number, string will be string, email will be email.
+But the tokenized version will look like a random value.
+Better see than read:
```
Comment (storojs72) [inline acra/security-controls/tokenisation/_index.md:13]:
```suggestion
It is better to see than to read:
```

File: acra/security-controls/tokenisation/_index.md [hunk 5]
```diff
@@ -2,3 +2,170 @@
...
+
+Tokenization is done by Acra Translator.
+Both protocols (gRPC and HTTP) provide the same functionality, but from different angles.
+You decide which one fits best for your application.
+
+### gRPC
+
+In order to use the translator gRPC API you have to take
+[api.proto](https://github.com/cossacklabs/acra-Q12021/blob/master/cmd/acra-translator/grpc_api/api.proto)
+and use either `service Tokenizator` or `service BulkProcessing` (with one one more `TokenizeRequest` inside).
```
Comment (storojs72) [inline acra/security-controls/tokenisation/_index.md:35]:
```suggestion
and use either `service Tokenizator` or `service BulkProcessing` (with one more `TokenizeRequest` inside).
```

File: acra/security-controls/tokenisation/_index.md [hunk 6]
```diff
@@ -2,3 +2,170 @@
...
+
+#### Golang example
+
+TODO add golang example
```
Comment (storojs72) [inline acra/security-controls/tokenisation/_index.md:39]: Do we have task for this?

File: acra/security-controls/tokenisation/_index.md [hunk 7]
```diff
@@ -2,3 +2,170 @@
...
+
+    eprintln!("DETOKENIZE RESPONSE: {:?}", &response.get_ref());
+
+    Ok(())
+}
+```
+
+### HTTP
+
+TODO add example of using curl or whatever
```
Comment (storojs72) [inline acra/security-controls/tokenisation/_index.md:171]: Same. Is there a task for it?

File: acra/security-controls/tokenisation/_index.md (hunk 6 above)
Comment (iamnotacake) [reply acra/security-controls/tokenisation/_index.md:39]: >mark it TODO for yourself and will add within this PR

this

**Review by Lagovas — APPROVED** [review]
lgtm

File: acra/security-controls/tokenisation/_index.md (hunk 7 above)
Comment (iamnotacake) [reply acra/security-controls/tokenisation/_index.md]: Added later in this PR

**Pull request merged**